bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,913 @@
+                    int id = Integer.parseInt(idStr);
+                    int state = Integer.parseInt(stateStr);
+                    QbusCO2 CO2 = new QbusCO2();
+                    if (!this.co2.containsKey(id)) {
+                        this.co2.put(id, CO2);
+                        CO2.setState(state);
+                    } else {
+                        CO2.setState(state);
+                    }
+                } else {
+                    logger.error("Error in json for CO2");
+                }
+
+            }
+        }
+    }
+
+    /**
+     * Get all the Thermostat outputs from the Qbus client
+     *
+     * @param data
+     */
+    private void cmdListThermostat(@Nullable List<Map<String, String>> outputs) {
+        if (outputs != null) {
+            for (Map<String, String> thermostat : outputs) {
+                String idStr = thermostat.get("id");
+                String measuredStr = thermostat.get("measured");
+                String setpointStr = thermostat.get("SetPoint");
+                String modeStr = thermostat.get("Mode");
+                if (idStr != null && measuredStr != null && setpointStr != null && modeStr != null) {
+                    int id = Integer.parseInt(idStr);
+                    Double measured = Double.valueOf(measuredStr);
+                    Double setpoint = Double.valueOf(setpointStr);
+                    Integer mode = Integer.valueOf(modeStr);
+                    QbusThermostat qThermostat = new QbusThermostat(idStr);
+                    if (!this.thermostat.containsKey(id)) {
+                        qThermostat.updateState(measured, setpoint, mode);
+                        qThermostat.setQComm(this);
+                        this.thermostat.put(id, qThermostat);
+                    } else {
+                        qThermostat.updateState(measured, setpoint, mode);
+                    }
+                } else {
+                    logger.error("Error in json for Thermostats");
+                }
+            }
+        }
+    }
+
+    /**
+     * Event on incoming Bistabiel/Timer/Mono/Interval updates
+     *
+     * @param data
+     */
+    private void updateBistabiel(List<Map<String, String>> output) {
+        for (Map<String, String> bistabiel : output) {
+            String idStr = bistabiel.get("id");
+            String stateStr = bistabiel.get("state");
+            if (idStr != null && stateStr != null) {
+                int id = Integer.parseInt(idStr);
+                int value1 = Integer.parseInt(stateStr);
+                QbusBistabiel Bistabiel = this.bistabiel.get(id);
+                if (Bistabiel != null) {
+                    if (!this.bistabiel.containsKey(id)) {
+                        logger.warn("Bistabiel in controller not known {}", id);
+                        return;
+                    }
+                    logger.info("Event execute bistabiel {} with state {}", id, value1);
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug.